bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/RecaptchaImpl.java

```diff
@@ -0,0 +1,108 @@
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.internal.models.v1.form;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.osgi.service.component.annotations.Reference;
+
+import com.adobe.aemds.guide.service.GuideException;
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.forms.core.components.internal.form.FormConstants;
+import com.adobe.cq.forms.core.components.models.form.Recaptcha;
+import com.adobe.cq.forms.core.components.util.AbstractFieldImpl;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { Recaptcha.class,
+        ComponentExporter.class },
+    resourceType = { FormConstants.RT_FD_FORM_RECAPTCHA_V1 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class RecaptchaImpl extends AbstractFieldImpl implements Recaptcha {
+
+    @Inject
+    private ResourceResolver resourceResolver;
+
+    private Resource resource;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Named("rcCloudServicePath")
+    protected String cloudServicePath;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    @Named("recaptchaSize")
+    protected String size;
+
+    public static final String RECAPTCHA_DEFAULT_DOMAIN = "https://www.recaptcha.net/";
+
+    public static final String RECAPTCHA_DEFAULT_URL = RECAPTCHA_DEFAULT_DOMAIN + "recaptcha/api.js";
+
+    private static final String RECAPTCHA_SITE_KEY = "siteKey";
+
+    private static final String RECAPTCHA_URI = "uri";
+
+    @Override
+    public String getCloudServicePath() {
+        return cloudServicePath;
+    }
+
+    @Override
+    public String getSize() {
+        return size;
+    }
+
+    @JsonIgnore
+    public Map<String, Object> getRecaptchaProperties() throws GuideException {
+
+        Map<String, Object> customCaptchaProperties = new LinkedHashMap<>();
+        String siteKey = null;
+        resource = resourceResolver.getResource(this.getPath());
+        if (resource != null) {
+            //todo: Add cloud configuration provider service
+        }
+        customCaptchaProperties.put(RECAPTCHA_SITE_KEY, siteKey);
```

Comment:
pls dont put anything in map until we support it completely. this will impact json.